### PR TITLE
fixed nonce too low errors

### PR DESCRIPTION
**Pull request by mihaib79**
lowered gas limit for Elrond transactions

### CODE REVIEW

File: bridge/elrond/client.go [hunk 1]
```diff
@@ -476,6 +482,9 @@ func (c *Client) sendTransaction(builder *txDataBuilder, cost uint64) (string, e
 	}
 
 	hash, err := c.proxy.SendTransaction(tx)
+	if err == nil {
```
Comment (alexandru-calinoiu) [inline bridge/elrond/client.go:485]: Initially I had a very similar implementation, maybe consider adding a test https://github.com/ElrondNetwork/elrond-eth-bridge/commit/115d4219547567878a37847a0591c1f14ec66541#diff-b0b337c55880463d4cc08d025f442af609bc7479ac061194a48900c28a6cbfb1L156

File: bridge/elrond/client.go [hunk 2]
```diff
@@ -73,35 +75,39 @@ func NewClient(config bridge.Config) (*Client, string, error) {
...
+	client := &Client{
 		proxy:         proxy,
 		bridgeAddress: config.BridgeAddress,
 		privateKey:    privateKey,
 		address:       address.AddressAsBech32String(),
-		nonce: func() (uint64, error) {
+		log:           log,
+	}
+
+	go func() {
```
Comment (alexandru-calinoiu) [inline bridge/elrond/client.go:86]: Maybe we should make this aware of the context, we do pass the context as it's needed by the eth sdk.

File: bridge/elrond/client.go [hunk 3]
```diff
@@ -17,13 +18,14 @@ import (
 )
 
 const (
-	ExecutionCost = 1000000000
+	ExecutionCost       = 250000000
```
Comment (alexandru-calinoiu) [inline bridge/elrond/client.go:21]: This won't work unfortunately based on costs here https://github.com/ElrondNetwork/sc-bridge-elrond/blob/main/docs/gas_costs.md

File: bridge/elrond/client.go (hunk 2 above)
Comment (iulianpascalau) [reply bridge/elrond/client.go:86]: agree, I would add a context with cancel function + an exported Close method as to be able to control this go routine.

File: bridge/elrond/client.go [hunk 4]
```diff
@@ -73,35 +75,39 @@ func NewClient(config bridge.Config) (*Client, string, error) {
...
-	}, addressString, nil
+	return client, addressString, nil
 }
 
 func (c *Client) GetPending(context.Context) *bridge.Batch {
 	c.log.Info("Elrond: Getting pending batch")
 	responseData, err := c.getCurrentBatch()
 	if err != nil {
-		c.log.Error(fmt.Sprintf("Error quering current batch: %q", err.Error()))
+		c.log.Error(fmt.Sprintf("Error querying current batch: %q", err.Error()))
```
Comment (iulianpascalau) [inline bridge/elrond/client.go:103]: we usually use this as following:
```GO
c.log.Error("error querying current batch", "error", err.Error())
```

File: bridge/elrond/client.go [hunk 5]
```diff
@@ -169,7 +175,7 @@ func (c *Client) ProposeSetStatus(_ context.Context, batch *bridge.Batch) {
 	if err != nil {
 		c.log.Error(err.Error())
 	}
-	c.log.Info(fmt.Sprintf("Elrond: Propsed status update with hash %s", hash))
+	c.log.Info(fmt.Sprintf("Elrond: Proposed status update with hash %s", hash))
```
Comment (iulianpascalau) [inline bridge/elrond/client.go:178]: c.log.Info("Elrond: Proposed status update", "hash", hash)

**Pull request merged**